lib/manager/regex/index.ts

```diff
@@ -45,6 +47,12 @@ export function extractPackageFile(
         }
       }
       dep.replaceString = String(matchResult[0]);
+      if (dep.registryUrl) {
+        if (url.parse(dep.registryUrl)) {
```

Comment:
`url.parse` is deprecated, and is likely to be removed in a new version of Node.js

> Use of the legacy `url.parse()` method is discouraged. Users should use the WHATWG `URL` API. Because the `url.parse()` method uses a lenient, non-standard algorithm for parsing URL strings, security issues can be introduced. Specifically, issues with host name spoofing and incorrect handling of usernames and passwords have been identified.

[See here for more information][nodejs].

It looks like the replacement is to do something like

```ts
let valid = false
try {
    new URL(dep.registryUrl);
    valid = true;
} catch {}
```

[nodejs]: https://nodejs.org/docs/latest-v15.x/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost

Reply:
We are using it in a few places so I'm going to accept it here but create a new issue to add something like a `isValidUrl()` function to our utils